dmoj/cli.py

```diff
@@ -202,17 +231,17 @@ def execute(self, line):
                 except Exception as io:
                     err = str(io)
             else:
-                src = []
+                file_suffix = '.' + language_id
                 try:
-                    while True:
-                        s = raw_input()
-                        if s.strip() == ':q':
-                            raise EOFError
-                        src.append(s)
-                except EOFError:  # Ctrl+D
-                    src = '\n'.join(src)
-                except Exception as io:
-                    err = str(io)
+                    with tempfile.NamedTemporaryFile(suffix=file_suffix) as temp:
+                        subprocess.call(['nano', temp.name])
```

Comment:
Aren't you supposed to keep the old code around for when `$EDITOR` is not defined?